reactioncommerce/reaction
# feat: Add flag to enable only IDP routes

Resolves #4852  
Impact:
Type: **feature**

## Issue
See #4852

### Goals for this PR:
- Ability to run the app in a mode that allows only routes from the Identity Provider Plugin

### Solution Overview
- New ENV var `IDENTITY_PROVIDER_MODE` that is used to filter published routes

### Details
- There are three values the ENV can be set to. Here's what they do:
  - Showing all routes (no restriction) (ENV value `all`)
  - Showing only Identity provider routes (ENV value `idp-only`)
  - Showing all routes excluding identity provider routes (ENV value `exclude-idp`)
- In the client router, the code logic that creates the routes array from package registry items is now wrapped as a function, so that it can be called with different args.


## Breaking changes
None


## Testing
Note: Make sure to remove existing `reaction` container after changing env value (e.g with `docker-compose down` and then up again).
1. Set IDENTITY_PROVIDER_MODE=all in your ENV.
Once app is started, we want to confirm that all routes are available. Home page should load the products and `/account/login` should show login form.

2. Set IDENTITY_PROVIDER_MODE=idp-only in your ENV.
Once app is started, we want to confirm that only IDP routes show. Viewing `/account/login` should show login form. Other usual pages (including home page) show Not Found.

2. Set IDENTITY_PROVIDER_MODE=exclude-idp in your ENV.
Once app is started, we want to confirm that IDP routes do not show. Viewing `/account/login` should not show up. Other usual pages (including home page) should show.
